resources/settings.xml

```diff
@@ -7,8 +7,8 @@
         <setting label="30901" type="action" id="info" action="RunScript(script.module.inputstreamhelper, info)"/>
         <setting label="30903" help="30904" type="bool" id="disabled" default="false"/>
         <setting label="30904" type="text" enable="false"/> <!-- disabled_warning -->
-        <setting label="30905" help="30906" type="action" id="install_widevine" action="RunScript(script.module.inputstreamhelper, widevine_install)" enable="eq(-2,false)" visible="!system.platform.android"/>
-        <setting label="30907" help="30908" type="action" id="remove_widevine" action="RunScript(script.module.inputstreamhelper, widevine_remove)" enable="eq(-3,false)" visible="!system.platform.android"/>
+        <setting label="30905" help="30906" type="action" id="install_widevine" action="RunScript(script.module.inputstreamhelper, widevine_install)" enable="eq(-2,false)" visible="!system.platform.android + System.HasAddon(inputstream.adaptive)"/>
```

Comment:
I would rather disable it if **inputstream.adaptive** is missing. So that it is clear that this functionality exists but is not accessible right now.

Reply:
I can't get this to work with the `eq(-2,false)` comparator. Maybe I'm doing something wrong, maybe Kodi doesn't support this.

Reply:
You are right, I cannot find any examples in existing add-ons that are doing this. I was expecting something like this to work, but you probably already tested this:

`enable="eq(-2,false) + System.HasAddon(inputstream.adaptive)"```

Reply:
While testing, it seems I cannot make it reload my settings easily. So testing requires to reinstall the **inputstreamhelper** addon :-/

Reply:
Despite this being pretty clear it should work: https://kodi.wiki/view/Add-on_settings#Conditions
Maybe this is related to the rewrite (and compatibility layer) not working correctly: https://forum.kodi.tv/showthread.php?tid=321370

Reply:
Most likely you need to add an invisible entry and reference it using `eq(-X,bool)` as well.

Reply:
No, I tried with the invisible entry, but I can't get the right value from `System.HasAddon(inputstream.adaptive)`

Reply:
If not, I don't mind merging this as-is. We can fix this if we move to the settings v2 interface (when we stop supporting Kodi v17)

Reply:
I changed it so settings are disabled when inputstream.adaptive is missing and settings get invisible when the "Disable InputStream Helper" switch is activated.